c/uca/sifter/unisifeggs.cpp

```diff
@@ -0,0 +1,433 @@
+#include <algorithm>
+#include <array>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <ranges>
+#include <string>
+#include <string_view>
+#include <vector>
+
+class CheckFailure {
+ public:
+  ~CheckFailure() {
+    std::terminate();
+  }
+  void operator&(std::ostream&) {}
+};
+
+#define CHECK(condition) \
+  (condition)            \
+      ? (void)0          \
+      : CheckFailure() & std::cerr << ("Check failed: " #condition "\n")
+
+class CodePointRange {
+ public:
+  constexpr CodePointRange(char32_t front, char32_t back)
+      : front_(front), back_(back) {
+    if (empty()) {
+      front_ = 1;
+      back_ = 0;
+    }
+  }
+
+  constexpr bool empty() const {
```

Comment:
Let's please call this isEmpty(). empty() sounds like a command to make it so.

Reply:
https://en.cppreference.com/w/cpp/named_req/Container.

Reply:
The more the C++ standard progresses, the less I like C++ ... :-(